classes/experiment_manager.php

```diff
@@ -145,12 +145,19 @@ private function invalidate_experiment_cache() {
 
     public function get_experiments() {
         $cache = cache::make('tool_abconfig', 'experiments');
-        return $cache->get('allexperiment');
+        $experiments = $cache->get('allexperiment');
+        if ($experiments != false) {
+            return $experiments;
+        } else {
+            return array();
+        }
     }
 
     public function get_active_request() {
         $experiments = self::get_experiments();
-
+        if ($experiments == false) {
```

Comment:
this seems redundant?

In fact wouldn't this whole patch be a much simpler fix if instead you just returned $data here:

https://github.com/catalyst/moodle-tool_abconfig/blob/master/classes/experiment_cache.php#L66

and deleted this line:

https://github.com/catalyst/moodle-tool_abconfig/blob/master/classes/experiment_cache.php#L63

Reply:
I removed the redundant checks. In terms of simplifying the patch, I think the cache_definition should be left as it is. In the case of the bug, changing the return in this instance does not resolve the issue, as this cache is never instantiated, and instead cache::make returns a cache_disabled object, which still returns a boolean false.

The cache definition was designed to follow the general cache spec, in which a false is returned if the key isnt found. https://docs.moodle.org/dev/Cache_API_-_Quick_reference#Get_a_key